plugins/common/network/api.ts

```diff
@@ -0,0 +1,5 @@
+export interface NetworkApi {
+  methods: {
+    version(): number;
```

Comment:
what does version represent here?

Reply:
It's the network id as described in the documentation : 
https://github.com/ethereum/wiki/wiki/JSON-RPC#net_version

Reply:
Would this serve as a redirection for a specific Provider that want to redirect  ethereum call 
 (sendTransaction,... ) to it?
In that case, could we rename this `NetworkProvider`, and the same for `Wallet`: `WalletProvider`?

Reply:
In this case we just have `provider` and that's it... Meaning that when we'll interact with a plugin we'll have to do `client.call("remixProvider", "sendAsync", "net_version")` for example.

Reply:
and the `remixProvider` will under the hood call this `NetworkApi`, am i right?

Reply:
What I meant is: I don't really see the point of calling it `NetworkProviderApi`. Either we separate `provider`, `wallet` and `network`. Or we have different king of `provider` (and in this case we don't have `NetworkApi`).
I just wanted to point out, with my example above, that the use a `provider` as it, might not be the best devX.

Reply:
So just to be sure we see the same thing: is the `NetworkApi` supposed to implement a subset of https://github.com/ethereum/wiki/wiki/JSON-RPC .

Reply:
from what i see:
`provider` is a global provider that propose interfaces which basically have all the interfaces of the web3 provider (network, transaction, account management, ...)
`network` aka `networkProvider` has only the network part.
`wallet` aka `walletProvider` has only the account management part.

A few use cases apply from here:
one want to have a really basic web3 provider, he can do that by having a `provider` plugin linked to a geth json rpc server.
one want to have a mixed provider, which use a `networkProvider` and a `walletProvider` from the current IDE instance.
one want to have a light provider, which will throw on all the functions except `sendRawTransaction` (`networkProvider`)

Reply:
This is quite how metamask provider engine works (with subProvider). That makes sense.

Reply:
Ok cool,
That's why I find useful to name it with the suffix `provider`.
`network` alone means a lot of stuff but `networkProvider` directly relates to the web3 provider, well that's a wording issue.

But now that I think of it, it might be that even a `networkProvider` might only need `sendAsync` as interface, then only the `provider` type is needed. perhaps we should jump on a call for discussing that.

Reply:
The problem with provider for me is that it's a low level API (with hex value), which is not the best developer experience. Maybe we should not try to have a strong opinion about what is network used for and just define the interface based on the JSON RPC API. Maybe network will call the provider, or maybe the provider will call the network, maybe they won't talk to each other, I think all options work.

Reply:
> define the interface based on the JSON RPC API

The input/output will be the same (hex value) if we stick to the JSON rpc ...

Reply:
sry, have to run, resuming this in a bit

Reply:
We have to stick with hex values if we want the provider to work with existing technologies. That's why I think it's nice for the `network` and `wallet` to be higher level APIs.